utils/writeable/src/try_writeable.rs

```diff
@@ -175,7 +175,7 @@ pub trait TryWriteable {
     }
 
     /// Returns a `&str` that matches the output of `write_to` (if it succeeds), if possible.
-    fn try_borrow(&self) -> Option<&str> {
+    fn writeable_borrow(&self) -> Option<&str> {
```

Comment:
Issue: I think `TryWriteable`'s method should be `try_writeable_borrow` and it should return `Result<Option<&str>, (Self::Error, Option<&str>)>`